tests/test_gene_analysis_integration.py

```diff
@@ -0,0 +1,172 @@
+import configparser
+import os
+
+import pytest
+import pandas as pd
+from typing import Dict
+from indra_cogex.client.neo4j_client import Neo4jClient
+from indra_cogex.analysis.gene_analysis import discrete_analysis, signed_analysis
+
+
+@pytest.fixture(scope="module")
+def neo4j_client() -> Neo4jClient:
+    client = Neo4jClient()
+    print(f"Neo4j client initialized: {client}")
+
+    # Attempt to set timeout if a method exists
+    if hasattr(client, 'set_timeout'):
+        client.set_timeout(60)
+    elif hasattr(client, 'driver') and hasattr(client.driver, 'set_timeout'):
+        client.driver.set_timeout(60)
+    else:
+        print("Warning: Unable to set timeout for Neo4jClient")
+
+    return client
+
+
+def get_neo4j_url():
+    # Try to read from config file
+    config = configparser.ConfigParser()
+    config_file = os.path.expanduser('~/.config/indra/config.ini')
+    if os.path.exists(config_file):
+        config.read(config_file)
+        if 'neo4j' in config and 'INDRA_NEO4J_URL' in config['neo4j']:
+            return config['neo4j']['INDRA_NEO4J_URL']
+
+    # If not found in config file, try environment variable
+    return os.getenv('INDRA_NEO4J_URL')
+
+
+# Print the Neo4j URL
+neo4j_url = get_neo4j_url()
+print(f"Neo4j Connection URL: {neo4j_url}")
+
+
+def test_neo4j_connection(neo4j_client: Neo4jClient):
+    try:
+        result = neo4j_client.query_tx("RETURN 1 as test")
+        assert result[0]['test'] == 1, "Failed to execute a simple query"
+        print("Successfully connected to Neo4j database")
+    except Exception as e:
+        pytest.fail(f"Failed to connect to Neo4j database: {str(e)}")
```

Comment:
This test could be replaced by
```suggestion
    assert neo4j_client.ping()
```

Reply:
Fixed in 95870169